From 218bbc555a37f9373fbb7f03c744eb65109d3470 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 9 Nov 2016 20:47:37 -0800
Subject: [PATCH 16/19] make test dir configurable

Upstream-Status: Pending

test maybe be run on target in cross-compile environment, and test dir
is not the compilation dir, so make it configurable

Signed-off-by: Roy Li <rongqing.li@windriver.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile.am  | 2 +-
 configure.ac | 7 +++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 229492a..e997d82 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -214,7 +214,7 @@ AM_CPPFLAGS = \
 	-DROOTLIBDIR=\"$(rootlibdir)\" \
 	-DROOTLIBEXECDIR=\"$(rootlibexecdir)\" \
 	-DROOTHOMEDIR=\"$(roothomedir)\" \
-	-DTEST_DIR=\"$(abs_top_srcdir)/test\" \
+	-DTEST_DIR=\"$(testdir)/test\" \
 	-I $(top_srcdir)/src \
 	-I $(top_builddir)/src/basic \
 	-I $(top_srcdir)/src/basic \
diff --git a/configure.ac b/configure.ac
index 1de0066..b12e320 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1535,6 +1535,11 @@ AC_ARG_WITH([roothomedir],
         [],
         [with_roothomedir=/root])
 
+AC_ARG_WITH([testdir],
+        AS_HELP_STRING([--with-testdir=DIR], [test file directory]),
+        [],
+        [with_testdir=${abs_top_srcdir}])
+
 AC_ARG_WITH([pamlibdir],
         AS_HELP_STRING([--with-pamlibdir=DIR], [Directory for PAM modules]),
         [],
@@ -1621,6 +1626,7 @@ AC_SUBST([pamconfdir], [$with_pamconfdir])
 AC_SUBST([rootprefix], [$with_rootprefix])
 AC_SUBST([rootlibdir], [$with_rootlibdir])
 AC_SUBST([roothomedir], [$with_roothomedir])
+AC_SUBST([testdir], [$with_testdir])
 
 AC_CONFIG_FILES([
         Makefile
@@ -1712,6 +1718,7 @@ AC_MSG_RESULT([
         lib dir:                           ${libdir}
         rootlib dir:                       ${with_rootlibdir}
         root home dir:                     ${with_roothomedir}
+        test dir:                          ${with_testdir}
         SysV init scripts:                 ${SYSTEM_SYSVINIT_PATH}
         SysV rc?.d directories:            ${SYSTEM_SYSVRCND_PATH}
         Build Python:                      ${PYTHON}
-- 
2.10.2

